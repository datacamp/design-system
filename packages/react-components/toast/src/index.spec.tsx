import '@testing-library/jest-dom/extend-expect';

import { render } from '@testing-library/react';
import React from 'react';

import { toast, ToastContainer } from '.';

jest.useFakeTimers();

describe('toast', () => {
  it('mounts a success toast', () => {
    const title = 'test toast title';
    const { getByRole, getByText } = render(<ToastContainer />);
    toast({ intent: 'success', title });
    jest.runAllTimers();
    const renderedToast = getByRole('alert');
    const titleElement = getByText(title) as HTMLElement;
    expect(renderedToast).toContainElement(titleElement);
    expect(renderedToast).toMatchSnapshot();
  });

  it('mounts an error toast', () => {
    const title = 'test toast title';
    const { getByRole, getByText } = render(<ToastContainer />);
    toast({ intent: 'error', title });
    jest.runAllTimers();
    const renderedToast = getByRole('alert');
    const titleElement = getByText(title) as HTMLElement;
    expect(renderedToast).toContainElement(titleElement);
    expect(renderedToast).toMatchSnapshot();
  });
});
