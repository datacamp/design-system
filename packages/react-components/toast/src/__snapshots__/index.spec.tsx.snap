// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`toast mounts a success toast 1`] = `
.emotion-5 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background: white;
  border-radius: 4px;
  box-shadow: 0px 0px 1px 0px rgba(5,25,45,0.3),0px 8px 12px -4px rgba(5,25,45,0.3);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  max-width: 512px;
  min-width: 250px;
  position: relative;
  z-index: 999;
}

.emotion-0 {
  background-color: #03ef62;
  border-radius: 4px 0px 0px 4px;
  height: 100%;
  position: absolute;
  width: 4px;
}

.emotion-1 {
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  margin-left: 16px;
}

.emotion-3 {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  margin: 12px;
}

.emotion-2 {
  -webkit-font-smoothing: antialiased;
  color: rgb(5,25,45);
  font-family: Arial;
  font-family: Studio-Feixen-Sans-Variable;
  font-style: normal;
  color: rgb(5,25,45);
  font-weight: 800;
  margin: 0;
  overflow: reset;
  white-space: normal;
  font-size: 16px;
  -webkit-letter-spacing: 0;
  -moz-letter-spacing: 0;
  -ms-letter-spacing: 0;
  letter-spacing: 0;
  line-height: 1.2;
}

*:not(style) ~ .emotion-2 {
  margin-top: 16px;
}

.emotion-4 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  aria-label: close;
  background: none;
  border: none;
  border-radius: 2px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  height: 20px;
  -webkit-box-pack: space-around;
  -webkit-justify-content: space-around;
  -ms-flex-pack: space-around;
  justify-content: space-around;
  margin-right: 12px;
  padding: 0;
  width: 20px;
}

.emotion-4:hover {
  background-color: #efefef;
}

<div
  class="Toastify__toast-body"
  role="alert"
>
  <div
    class="emotion-5"
  >
    <div
      class="emotion-0"
    />
    <svg
      aria-hidden="false"
      class="emotion-1"
      height="18"
      role="img"
      viewBox="0 0 18 18"
      width="18"
    >
      <title>
        success
      </title>
      <path
        d="M9 18A9 9 0 119 0a9 9 0 010 18zm2.326-11.96l-3.355 4.143L6.63 8.627a1 1 0 10-1.514 1.307l2.124 2.461a1 1 0 001.534-.024l4.108-5.073a1 1 0 00-1.555-1.259z"
        fill="#03ef62"
        fill-rule="evenodd"
      />
    </svg>
    <div
      class="emotion-3"
    >
      <h4
        class="emotion-2"
      >
        test toast title
      </h4>
    </div>
    <button
      aria-label="Close Toast Notification"
      class="emotion-4"
      type="button"
    >
      <svg
        aria-hidden="true"
        height="12"
        role="img"
        viewBox="0 0 18 18"
        width="12"
      >
        <path
          d="M9.005 7.625l4.83-4.83a.976.976 0 011.38 1.38l-4.83 4.83 4.82 4.82a.976.976 0 11-1.38 1.38l-4.82-4.82-4.83 4.83a.976.976 0 01-1.38-1.38l4.83-4.83-4.84-4.84a.976.976 0 111.38-1.38l4.84 4.84z"
          fill="currentColor"
          fill-rule="evenodd"
        />
      </svg>
    </button>
  </div>
</div>
`;

exports[`toast mounts an error toast 1`] = `
.emotion-5 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background: white;
  border-radius: 4px;
  box-shadow: 0px 0px 1px 0px rgba(5,25,45,0.3),0px 8px 12px -4px rgba(5,25,45,0.3);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  max-width: 512px;
  min-width: 250px;
  position: relative;
  z-index: 999;
}

.emotion-1 {
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  margin-left: 16px;
}

.emotion-3 {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  margin: 12px;
}

.emotion-2 {
  -webkit-font-smoothing: antialiased;
  color: rgb(5,25,45);
  font-family: Arial;
  font-family: Studio-Feixen-Sans-Variable;
  font-style: normal;
  color: rgb(5,25,45);
  font-weight: 800;
  margin: 0;
  overflow: reset;
  white-space: normal;
  font-size: 16px;
  -webkit-letter-spacing: 0;
  -moz-letter-spacing: 0;
  -ms-letter-spacing: 0;
  letter-spacing: 0;
  line-height: 1.2;
}

*:not(style) ~ .emotion-2 {
  margin-top: 16px;
}

.emotion-4 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  aria-label: close;
  background: none;
  border: none;
  border-radius: 2px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  height: 20px;
  -webkit-box-pack: space-around;
  -webkit-justify-content: space-around;
  -ms-flex-pack: space-around;
  justify-content: space-around;
  margin-right: 12px;
  padding: 0;
  width: 20px;
}

.emotion-4:hover {
  background-color: #efefef;
}

.emotion-0 {
  background-color: #ff5400;
  border-radius: 4px 0px 0px 4px;
  height: 100%;
  position: absolute;
  width: 4px;
}

<div
  class="Toastify__toast-body"
  role="alert"
>
  <div
    class="emotion-5"
  >
    <div
      class="emotion-0"
    />
    <svg
      aria-hidden="false"
      class="emotion-1"
      height="18"
      role="img"
      viewBox="0 0 18 18"
      width="18"
    >
      <title>
        error
      </title>
      <path
        d="M9 18A9 9 0 119 0a9 9 0 010 18zm1.414-9l2.122-2.121a1 1 0 10-1.415-1.415L9 7.586 6.879 5.464A1 1 0 005.464 6.88L7.586 9l-2.122 2.121a1 1 0 001.415 1.415L9 10.414l2.121 2.122a1 1 0 001.415-1.415L10.414 9z"
        fill="#ff5400"
        fill-rule="evenodd"
      />
    </svg>
    <div
      class="emotion-3"
    >
      <h4
        class="emotion-2"
      >
        test toast title
      </h4>
    </div>
    <button
      aria-label="Close Toast Notification"
      class="emotion-4"
      type="button"
    >
      <svg
        aria-hidden="true"
        height="12"
        role="img"
        viewBox="0 0 18 18"
        width="12"
      >
        <path
          d="M9.005 7.625l4.83-4.83a.976.976 0 011.38 1.38l-4.83 4.83 4.82 4.82a.976.976 0 11-1.38 1.38l-4.82-4.82-4.83 4.83a.976.976 0 01-1.38-1.38l4.83-4.83-4.84-4.84a.976.976 0 111.38-1.38l4.84 4.84z"
          fill="currentColor"
          fill-rule="evenodd"
        />
      </svg>
    </button>
  </div>
</div>
`;
